tests/UnitTests/Foundation/NSIndexPathTests.mm

```diff
@@ -0,0 +1,167 @@
+//******************************************************************************
+//
+// Copyright (c) Microsoft. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#include <TestFramework.h>
+#import <Foundation/Foundation.h>
+#import <Starboard/SmartTypes.h>
+
+TEST(NSIndexPath, Init) {
+    NSIndexPath* path = [[[NSIndexPath alloc] init] autorelease];
+    ASSERT_NE(nil, path);
+    EXPECT_EQ(0, [path length]);
+    EXPECT_EQ(NSNotFound, [path indexAtPosition:0]);
+}
+
+TEST(NSIndexPath, InitWithIndex) {
+    NSIndexPath* path = [[[NSIndexPath alloc] initWithIndex:5] autorelease];
```

Comment:
same at other places.

---
In reply to: [107549224](https://github.com/Microsoft/WinObjC/pull/2279#discussion_r107549224) [](ancestors = 107549224)